File: ria/java/gof/src/main/java/com/gof/pattern/factoryMethod/after/DefaultShipFactory.java
package com.gof.pattern.factoryMethod.after;

/** 자바 8을 사용하는 경우, 추상 클래스로 구현 **/
public abstract class DefaultShipFactory implements ShipFactory{

    // 기존 private 메서드 오버라이딩
    @Override
    public void sendEmailTo(String email, Ship ship) {
        System.out.println(ship.getName() + " 다 만들었습니다.");
    }
}

File: zeri/gof/src/main/kotlin/com/gof/pattern/factorymethod/javatokotlin/after/DefaultShipFactory.kt
package com.gof.pattern.factorymethod.javatokotlin.after

import com.gof.pattern.factorymethod.javatokotlin.after.Ship
import com.gof.pattern.factorymethod.javatokotlin.after.ShipFactory

abstract class DefaultShipFactory : ShipFactory {
    override fun sendEmailTo(name: String, ship: Ship) {
        println("$name 다 만들었습니다.")
    }
}
